### PR TITLE
GPU remote tensor tests update. Get rid of API 1.0

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - GPU remote tensor tests update. Get rid of API 1.0

### Tickets:
 - [CVS-126432](https://jira.devtools.intel.com/browse/CVS-126432)

### CODE REVIEW

**Comment by olpipi** [comment]
No need to update tests/functional/remote_blob_tests/cldnn_remote_blob_tests.cpp because these are legacy tests for api 1.0. We have already new analogue tests for api 2.0

**Review by p-durandin — APPROVED** [review]

File: src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp [hunk 1]
```diff
@@ -8,14 +8,14 @@
 #include <tuple>
 #include <memory>
 
-#include <ie_compound_blob.h>
```
Comment (vurusovs) [inline src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp:11]: Will this file be ported to API2.0?

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp (hunk 1 above)
Comment (olpipi) [reply src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp:11]: removed

Comment (vurusovs) [reply src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp:11]: Please, clean the file from InferenceEngine usage (e.g. https://github.com/openvinotoolkit/openvino/blob/fd899ed71943ee31bfa80093f8d6ae8b2ec5c60d/src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp#L206)

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp (hunk 1 above)
Comment (olpipi) [reply src/plugins/intel_gpu/tests/functional/remote_blob_tests/dx11_remote_ctx_test.cpp:11]: I refactored this test to use new API. Need to wait for CI

**Review by vurusovs — APPROVED** [review]

**Pull request merged**